docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -52,52 +52,130 @@ const theme = createTheme({
 
 {{"demo": "GlobalThemeOverride.js"}}
 
-Each component is composed of several different parts.
-These parts correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
-You can use these classes inside the `styleOverrides` key to modify the corresponding parts of the component.
+Each component is composed of several different slots.
+These slots correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
+You can use these classes inside the `styleOverrides` key to modify the corresponding slots of the component.
+
+### Overrides based on props
+
+Each slot of the component's `styleOverrides` supports a `variants` key. Variants are sets of overrides applied to the slot when the specified props match.
+
+The definitions are specified in an array. Make sure the styles that should take precedence are listed last.
+
+**Example 1**: Overriding styles based on existing props. To increase the border thickness of the `outlined` Card, use this following snippet:
+
+```js
+const theme = createTheme({
+  components: {
+    MuiCard: {
+      styleOverrides: {
+        root: {
+          variants: [
+            {
+              props: { variant: 'outlined' },
+              style: {
+                borderWidth: '3px',
+              },
+            },
+          ],
+        },
+      },
+    },
+  },
+});
+```
+
+**Example 2**: Overriding styles based on new values. If your project needs a new Button's variant, use the following snippet:
```

Comment:
For example, this doesn't feel like an override to me—it's a completely new thing that we're adding to the component's surface area.

Reply:
It's hard to justify. Even though it's about adding new value, it's potentially overriding the default styles of the Button.